portfolio/src/main/java/com/google/sps/data/MergeBooks.java

```diff
@@ -0,0 +1,174 @@
+package com.google.sps.data;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.sps.data.Book;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+public class MergeBooks {
+  /**
+   * This method merges two books that have the same title
+   * and ISBN and returns a new book combined from both of them
+   * @param original: the original book
+   * @param addition: the book with additional info you
+   * want to add
+   * The order of the params does not matter
+   */
+  public static Book merge(Book original, Book addition) {
+    validate(original);
+    validate(addition);
+    String isbn = getNonNullString(original.isbn(),
+        addition.isbn()); // One of the ISBNs will be null so you only need to have one valid isbn
+    validateISBN(isbn);
+    Preconditions.checkArgument(
+        original.title() == addition.title(), "The books must be of the same title");
```

Comment:
May be safer to use .equals() for string comparison
https://www.geeksforgeeks.org/difference-equals-method-java/